File: Lesson.03 - Design Patterns - Structural/Bridge/POP3.java
public class POP3 implements EmailProtocol {
    @Override
    public void initialize() {
        System.out.println("POP3: Initializing post office protocol");
    }

    @Override
    public void sendEmail(String message) {
        System.out.println("POP3: Sending email: " + message);
    }

    @Override
    public void shutdown() {
        System.out.println("POP3: Shutting down protocol");
    }
}

File: Structural - Exercises/Solutions-Leonie/Bridge/protocols/pop3.py
from protocols.protocol_interface import ProtocolInterface

class POP3Protocol(ProtocolInterface):
    def initialize(self):
        print("POP3: Initializing email protocol")

    def send_email(self, message: str):
        print(f"POP3: Sending email: {message}")
        self.provider.process_email(message)

    def shutdown(self):
        print("POP3: Shutting down protocol")
